dataframe-jdbc/src/main/kotlin/org/jetbrains/kotlinx/dataframe/io/readJdbc.kt

```diff
@@ -0,0 +1,635 @@
+            val tableColumns = getTableColumnsMetadata(rs)
+            val data = fetchAndConvertDataFromResultSet(tableColumns, rs, dbType, DEFAULT_LIMIT)
+
+            logger.debug { "SQL query executed successfully. Converting data to DataFrame." }
+
+            return data.toDataFrame()
+        }
+    }
+}
+
+/**
+ * Reads the data from a ResultSet and converts it into a DataFrame.
+ *
+ * @param [resultSet] the ResultSet containing the data to read.
+ * @param [dbType] the type of database that the ResultSet belongs to.
+ * @return the DataFrame generated from the ResultSet data.
+ */
+public fun DataFrame.Companion.readResultSet(resultSet: ResultSet, dbType: DbType): AnyFrame {
+    return readResultSet(resultSet, dbType, DEFAULT_LIMIT)
+}
+
+/**
+ * Reads the data from a ResultSet and converts it into a DataFrame.
+ *
+ * @param [resultSet] the ResultSet containing the data to read.
+ * @param [dbType] the type of database that the ResultSet belongs to.
+ * @param [limit] the maximum number of rows to read from the ResultSet.
+ * @return the DataFrame generated from the ResultSet data.
+ */
+public fun DataFrame.Companion.readResultSet(resultSet: ResultSet, dbType: DbType, limit: Int): AnyFrame {
+    val tableColumns = getTableColumnsMetadata(resultSet)
+    val data = fetchAndConvertDataFromResultSet(tableColumns, resultSet, dbType, limit)
+    return data.toDataFrame()
+}
+
+/**
+ * Reads the data from a ResultSet and converts it into a DataFrame.
+ *
+ * @param [resultSet] the ResultSet containing the data to read.
+ * @param [connection] the connection to the database (it's required to extract the database type).
+ * @return the DataFrame generated from the ResultSet data.
+ */
+public fun DataFrame.Companion.readResultSet(resultSet: ResultSet, connection: Connection): AnyFrame {
+    return readResultSet(resultSet, connection, DEFAULT_LIMIT)
+}
+
+/**
+ * Reads the data from a ResultSet and converts it into a DataFrame.
+ *
+ * @param [resultSet] the ResultSet containing the data to read.
+ * @param [connection] the connection to the database (it's required to extract the database type).
+ * @param [limit] the maximum number of rows to read from the ResultSet.
+ * @return the DataFrame generated from the ResultSet data.
+ */
+public fun DataFrame.Companion.readResultSet(resultSet: ResultSet, connection: Connection, limit: Int): AnyFrame {
+    val url = connection.metaData.url
+    val dbType = extractDBTypeFromURL(url)
+
+    return readResultSet(resultSet, dbType, limit)
+}
+
+/**
+ * Reads all non-system tables from a database and returns them as a list of data frames
+ * using the provided database configuration.
+ *
+ * @param [dbConfig] the database configuration to connect to the database, including URL, user, and password.
+ * @return a list of [AnyFrame] objects representing the non-system tables from the database.
+ */
+public fun DataFrame.Companion.readAllTables(dbConfig: DatabaseConfiguration): List<AnyFrame> {
```

Comment:
readAllSqlTables maybe? This is defined on the entire DataFrame.Companion scope

Reply:
Also, this doesn't return the names of the tables right? may be better to return a map

Reply:
Another suggestion for the future: better notebooks support. We can generate a type safe wrapper for this map. From user perspective it'll be:
```
val tables = DataFrame.readAllTables(...)
```
next cell (each table has schema):
```
tables.employee
tables.tracks
...
```

Reply:
I explored API after proposed changes and found that for meta information better to add separate methods or use methods returning a data schema

When I changed it to map, it overcomplicated a path to the dataframes

But I also thought, that it's great to have a field name for dataframes and one global space of all created dataframes to search between them by name

Reply:
One global space of all created dataframes would break the functional aspect of Kotlin DataFrame. They are immutable and any modification makes a new instance.
We could have a DataFrame Collection of some kind, but that would not be much different than a Map, or be able to name DataFrames.
Naming a DataFrame is not that far-fetched, since essentially, a DataFrame and ColumnGroup are the same and ColumnGroups can be named too. That said, currently variable names are used to refer to a DataFrame. There would be a mismatch when people would write: `val dataFrameName = dataFrameOf(...) named "otherDataFrameName"`.

Maybe in the meantime, you could return a DataFrame containing a "name: ValueColumn<String>" and "df: FrameColumn" column, which, especially in notebooks, would be a clear way for users to access multiple dataframes using a single one.